Behaviors/GroupBoxBehaviors.cs

```diff
@@ -0,0 +1,90 @@
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Media;
+using Thingie.WPF.Converters;
+
+namespace Thingie.WPF.Behaviors
+{
+    public class GroupBoxBehaviors
+    {
+        public static readonly DependencyProperty HideOnChildrenNotVisibleProperty = DependencyProperty.RegisterAttached("HideOnChildrenNotVisible", typeof(bool), typeof(GroupBoxBehaviors), new PropertyMetadata(default(bool), OnHideOnChildrenNotVisibleChanged));
+        public static void SetHideOnChildrenNotVisible(GroupBox element, bool value) { element.SetValue(HideOnChildrenNotVisibleProperty, value); }
+        public static bool GetHideOnChildrenNotVisible(GroupBox element) { return (bool)element.GetValue(HideOnChildrenNotVisibleProperty); }
+
+        private static void OnHideOnChildrenNotVisibleChanged(DependencyObject obj, DependencyPropertyChangedEventArgs args)
+        {
+            GroupBox groupBox = obj as GroupBox;
+
+            if (groupBox == null)
+                return;
+
+            if (groupBox.IsLoaded)
+            {
+                var multiBinding = new MultiBinding
+                {
+                    Converter = new MultiVisibilityToVisibilityConverter()
+                };
+
+                if ((bool)args.NewValue)
+                {
+                    foreach (var child in FindVisualChildren<Button>(groupBox))
+                    {
+                        multiBinding.Bindings.Add(new Binding("Visibility") { Mode = BindingMode.OneWay, Source = child });
+                    }
+                    BindingOperations.SetBinding(groupBox, UIElement.VisibilityProperty, multiBinding);
+                } 
+                else
+                {
+                    BindingOperations.ClearBinding(groupBox, UIElement.VisibilityProperty);
+                }
+                
+            }
+            else
+            {
+                groupBox.Loaded += (sender, eventArgs) =>
+                {
+                    var multiBinding = new MultiBinding
+                    {
+                        Converter = new MultiVisibilityToVisibilityConverter()
+                    };
+
+                    if ((bool)args.NewValue)
+                    {
+                        foreach (var child in FindVisualChildren<Button>(groupBox))
+                        {
+                            multiBinding.Bindings.Add(new Binding("Visibility") { Mode = BindingMode.OneWay, Source = child });
+                        }
+                        BindingOperations.SetBinding(groupBox, UIElement.VisibilityProperty, multiBinding);
+                    }
+                    else
+                    {
+                        BindingOperations.ClearBinding(groupBox, UIElement.VisibilityProperty);
+                    }
+                };
+            }
+
+        }
+
+        public static IEnumerable<T> FindVisualChildren<T>(DependencyObject depObj) where T : DependencyObject
```

Comment:
This will not work if the groupbox has subgroups. If the generic parameter is `<GroupBox>` instead of `<Button>` it will return the original GroupBox whose visibility is true and the groupbox will always be visible regardless of the visibility of its children.

Reply:
How? Doesn't the `VisualTreeHelper.GetChildrenCount(depObj)` method return only the number of children of the element (in your example, the number of children of the specified GroupBox)? And consequently, the `VisualTreeHelper.GetChild(depObj, i)` method returns the child GroupBox at the specified index.

Reply:
Ah, I misread your code, I thought it was standard recursive descent, but it... isn't. Are you sure you're using recursion correctly here? Two loops + recursion seem quite strange.

Reply:
Also, why use recursion here? If the groupbox has a control with visible=collapsed, should it care if that invisible child control has visible child controls?

Reply:
Let me check on an example with nested GroupBoxes.

Reply:
Yeah actually you are right, no need to test. The recursion doesn't make any sense here, I'll remove it.

Reply:
I tested this again and the recursion is needed. Contrary to my belief, `VisualTreeHelper.GetChildrenCount` only returns the number of immediate children. So the first child is going to be a Grid element, then the Border, then the ContentPresenter and so on. I think we forgot that the GroupBox element has multiple elements before the actual design we write(the Buttons and so on).

Reply:
Updated the method, let me know what you think.